tools/data/activitynet/convert_proposal_format.py

```diff
@@ -87,46 +91,87 @@ def dump_classifier_input(video_idx, video_id, num_frames, fps, gts, proposals,
                 f'{proposals[i][0]} {proposals[i][1]}\n')
 
 
-data_path = '../../../data/ActivityNet'
-result_path = '../../..'
-proposal_generator_output_file = f'{result_path}/results.json'
-action_classifier_input_file = f'{result_path}/anet_val_classifier_input.txt'
-activity_index_file = f'{data_path}/anet_activity_indexes_val.txt'
-ann_file = f'{data_path}/anet_anno_val.json'
-
-action_classifier_input = open(action_classifier_input_file, 'w')
-# The activity index file is constructed according to
-# 'https://github.com/activitynet/ActivityNet/blob/master/Evaluation/eval_classification.py'
-activity_index, class_idx = {}, 0
-for line in open(activity_index_file).readlines():
-    activity_index[line.strip()] = class_idx
-    class_idx += 1
-
-video_infos = load_annotations(ann_file)
-ground_truth = import_ground_truth(video_infos, activity_index)
-proposal, num_proposals = import_proposals(
-    mmcv.load(proposal_generator_output_file)['results'])
-video_idx = 0
-
-for video_info in video_infos:
-    video_id = video_info['video_name'][2:]
-    num_frames = video_info['duration_frame']
-    fps = video_info['fps']
-    tiou, t_overlap_self = pairwise_temporal_iou(
-        proposal[video_id][:, :2].astype(float),
-        ground_truth[video_id][:, :2].astype(float),
-        overlap_self=True)
-
-    dump_classifier_input(video_idx, video_id, num_frames, fps,
-                          ground_truth[video_id], proposal[video_id], tiou,
-                          t_overlap_self, action_classifier_input)
-    video_idx += 1
-action_classifier_input.close()
+def parse_args():
+    parser = argparse.ArgumentParser(description='convert proposal format')
+    parser.add_argument(
+        '--data-path',
+        type=str,
+        default='../../../data/ActivityNet',
+        help='path of activitynet data')
+    parser.add_argument(
+        '--result-path',
+        type=str,
+        default='../../..',
+        help='path of proposal result')
+    parser.add_argument(
+        '--ann-file',
+        type=str,
+        default='anet_anno_val.json',
+        help='name of annotation file')
+    parser.add_argument(
+        '--activity-index-file',
+        type=str,
+        default='anet_activity_indexes_val.txt',
+        help='name of activity index file')
+    parser.add_argument(
+        '--proposal-file',
+        type=str,
+        default='results.json',
+        help=
+        'name of proposal file, which is the output of proposal generator (BMN)'
+    )
+    parser.add_argument(
+        '--formatted-proposal-file',
+        type=str,
+        default='anet_val_formatted_proposal.txt',
+        help=
+        'name of formatted proposal file, which is the input of action classifier (SSN)'
+    )
+    args = parser.parse_args()
+
+    return args
+
+
+if __name__ == '__main__':
+    args = parse_args()
+    proposal_file = osp.join(args.result_path, args.proposal_file)
```

Comment:
Do not need to split it into dir and filename. Just use `proposal_file` as the path.